### PR TITLE
#1030 Add schema updates

### LINKED ISSUE #1030: GQL Update #69
Need to do an update to codegen and queries

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1030 

- The backend is still not working for me - still unable to integrate nsj's so somewhat limited testing
- I just added an nsj in sqlite directly for a name/store so do some mutations

### CODE REVIEW

File: packages/common/src/types/schema.ts [hunk 1]
```diff
@@ -159,6 +159,7 @@ export type BatchRequestRequisitionResponse = {
 };
 
 export type BatchStocktakeInput = {
+  continueOnError?: InputMaybe<Scalars['Boolean']>;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:162]: No change needed

File: packages/common/src/types/schema.ts [hunk 2]
```diff
@@ -167,20 +168,8 @@ export type BatchStocktakeInput = {
...
-  insertStocktakeLines?: Maybe<Array<InsertStocktakeLineResponseWithId>>;
-  insertStocktakes?: Maybe<Array<InsertStocktakeResponseWithId>>;
-  updateStocktakeLines?: Maybe<Array<UpdateStocktakeLineResponseWithId>>;
-  updateStocktakes?: Maybe<Array<UpdateStocktakeResponseWithId>>;
-};
-
-export type BatchStocktakeResponsesWithErrors = {
-  __typename: 'BatchStocktakeResponsesWithErrors';
+export type BatchStocktakeResponse = {
+  __typename: 'BatchStocktakeResponse';
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:172]: change needed for this in stocktake/api

File: packages/common/src/types/schema.ts [hunk 3]
```diff
@@ -221,6 +210,11 @@ export type CannotEditRequisition = AddFromMasterListErrorInterface & CreateRequ
   description: Scalars['String'];
 };
 
+export type CannotEditStocktake = DeleteStocktakeErrorInterface & DeleteStocktakeLineErrorInterface & InsertStocktakeLineErrorInterface & UpdateStocktakeErrorInterface & UpdateStocktakeLineErrorInterface & {
+  __typename: 'CannotEditStocktake';
+  description: Scalars['String'];
+};
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:216]: no change needed

File: packages/common/src/types/schema.ts [hunk 4]
```diff
@@ -462,34 +456,41 @@ export type DeleteResponse = {
...
 
+export type DeleteStocktakeError = {
+  __typename: 'DeleteStocktakeError';
+  error: DeleteStocktakeErrorInterface;
+};
+
+export type DeleteStocktakeErrorInterface = {
+  description: Scalars['String'];
+};
+
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:467]: no change needed

File: packages/common/src/types/schema.ts [hunk 5]
```diff
@@ -462,34 +456,41 @@ export type DeleteResponse = {
   id: Scalars['String'];
 };
 
+export type DeleteStocktakeError = {
+  __typename: 'DeleteStocktakeError';
+  error: DeleteStocktakeErrorInterface;
+};
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:462]: no change needed

File: packages/common/src/types/schema.ts [hunk 6]
```diff
@@ -462,34 +456,41 @@ export type DeleteResponse = {
...
+
 export type DeleteStocktakeInput = {
   id: Scalars['String'];
 };
 
-export type DeleteStocktakeLineInput = {
-  id: Scalars['String'];
+export type DeleteStocktakeLineError = {
+  __typename: 'DeleteStocktakeLineError';
+  error: DeleteStocktakeLineErrorInterface;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:474]: no change needed

File: packages/common/src/types/schema.ts [hunk 7]
```diff
@@ -462,34 +456,41 @@ export type DeleteResponse = {
...
 };
 
-export type DeleteStocktakeLineNode = {
-  __typename: 'DeleteStocktakeLineNode';
+export type DeleteStocktakeLineErrorInterface = {
+  description: Scalars['String'];
+};
+
+export type DeleteStocktakeLineInput = {
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:481]: no change needed

File: packages/common/src/types/schema.ts [hunk 8]
```diff
@@ -462,34 +456,41 @@ export type DeleteResponse = {
...
   id: Scalars['String'];
 };
 
-export type DeleteStocktakeLineResponse = DeleteStocktakeLineNode;
+export type DeleteStocktakeLineResponse = DeleteResponse | DeleteStocktakeLineError;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:485]: no change needed

File: packages/common/src/types/schema.ts [hunk 9]
```diff
@@ -462,34 +456,41 @@ export type DeleteResponse = {
...
 };
 
-export type DeleteStocktakeNode = {
-  __typename: 'DeleteStocktakeNode';
-  /** The id of the deleted stocktake */
-  id: Scalars['String'];
-};
-
-export type DeleteStocktakeResponse = DeleteStocktakeNode;
+export type DeleteStocktakeResponse = DeleteResponse | DeleteStocktakeError;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:493]: no change needed

File: packages/common/src/types/schema.ts [hunk 10]
```diff
@@ -521,6 +522,12 @@ export type EqualFilterItemTypeInput = {
   notEqualTo?: InputMaybe<ItemNodeType>;
 };
 
+export type EqualFilterReportCategoryInput = {
+  equalAny?: InputMaybe<Array<ReportCategory>>;
+  equalTo?: InputMaybe<ReportCategory>;
+  notEqualTo?: InputMaybe<ReportCategory>;
+};
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:529]: no change needed

File: packages/common/src/types/schema.ts [hunk 11]
```diff
@@ -928,13 +935,21 @@ export type FullQuery = {
...
   /** Query omSupply "name" entries */
   names: NamesResponse;
-  /** Query omSupply "locations" entries */
+  /**
+   * Creates a printed report.
+   *
+   * All details about the report, e.g. the output format, are specified in the report definition
+   * which is referred to by the report_id.
+   * The printed report can be retrieved from the `/files` endpoint using the returned file id.
+   */
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:944]: no change needed

File: packages/common/src/types/schema.ts [hunk 12]
```diff
@@ -928,13 +935,21 @@ export type FullQuery = {
...
   printReport: PrintReportResponse;
   /**
    * Retrieves a new auth bearer and refresh token
    * The refresh token is returned as a cookie
    */
   refreshToken: RefreshTokenResponse;
+  /** Queries a list of available reports */
+  reports: ReportsResponse;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:952]: no change needed

File: packages/common/src/types/schema.ts [hunk 13]
```diff
@@ -1018,6 +1033,14 @@ export type FullQueryPrintReportArgs = {
 };
 
 
+export type FullQueryReportsArgs = {
+  filter?: InputMaybe<ReportFilterInput>;
+  page?: InputMaybe<PaginationInput>;
+  sort?: InputMaybe<Array<ReportSortInput>>;
+  storeId: Scalars['String'];
+};
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1041]: no change needed

File: packages/common/src/types/schema.ts [hunk 14]
```diff
@@ -1350,6 +1373,15 @@ export type InsertStocktakeInput = {
...
 
+export type InsertStocktakeLineError = {
+  __typename: 'InsertStocktakeLineError';
+  error: InsertStocktakeLineErrorInterface;
+};
+
+export type InsertStocktakeLineErrorInterface = {
+  description: Scalars['String'];
+};
+
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1384]: no change needed

File: packages/common/src/types/schema.ts [hunk 15]
```diff
@@ -1366,7 +1398,7 @@ export type InsertStocktakeLineInput = {
   stocktakeId: Scalars['String'];
 };
 
-export type InsertStocktakeLineResponse = StocktakeLineNode;
+export type InsertStocktakeLineResponse = InsertStocktakeLineError | StocktakeLineNode;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1401]: no change needed

File: packages/common/src/types/schema.ts [hunk 16]
```diff
@@ -1469,9 +1501,10 @@ export type InvoiceLineNode = {
   note?: Maybe<Scalars['String']>;
   numberOfPacks: Scalars['Int'];
   packSize: Scalars['Int'];
+  pricing: PricingNode;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1504]: this is a bit weird but might be my fault. I suggested that the pricing structure for entities be consistent. Trying to say that `InvoiceNode` had an `InvoicePricingNode` which had ` {total, subtotal, stockTotal, serviceTotal, tax}` but on `InvoiceLineNode` it was all flattened.

So in addition to the flattened fields there has been the same object added on an invoice line (where having serviceTotal, stockTotal and totalTotal fields is not helpful)

This required adding the pricing node where `InvoiceLineNode` was being used (`useNextItem.test`) - though that test needs tidying in general

File: packages/common/src/types/schema.ts [hunk 17]
```diff
@@ -1469,9 +1501,10 @@ export type InvoiceLineNode = {
...
   sellPricePerPack: Scalars['Float'];
   stockLine?: Maybe<StockLineNode>;
-  textPercentage?: Maybe<Scalars['Float']>;
+  taxPercentage?: Maybe<Scalars['Float']>;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1507]: this is also in the pricing field

File: packages/common/src/types/schema.ts [hunk 18]
```diff
@@ -1512,6 +1545,12 @@ export type InvoiceNode = {
   status: InvoiceNodeStatus;
   theirReference?: Maybe<Scalars['String']>;
   type: InvoiceNodeType;
+  /**
+   * User that last edited invoice, if user is not found in system default unknow user is returned
+   * Null is returned for transfers, where inbound has not been edited yet
+   * Null is also returned for system created invoices like inventory adjustments
+   */
+  user?: Maybe<UserNode>;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1553]: no change needed

File: packages/common/src/types/schema.ts [hunk 19]
```diff
@@ -1934,6 +1963,17 @@ export type PaginationInput = {
...
+  __typename: 'PricingNode';
+  serviceTotalAfterTax: Scalars['Float'];
+  serviceTotalBeforeTax: Scalars['Float'];
+  stockTotalAfterTax: Scalars['Float'];
+  stockTotalBeforeTax: Scalars['Float'];
+  taxPercentage?: Maybe<Scalars['Float']>;
+  totalAfterTax: Scalars['Float'];
+  totalBeforeTax: Scalars['Float'];
+};
+
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1976]: no change needed

File: packages/common/src/types/schema.ts [hunk 20]
```diff
@@ -1993,6 +2033,50 @@ export type RegisteredUser = {
   username: Scalars['String'];
 };
 
+export enum ReportCategory {
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:2036]: no change needed for report stuff

File: packages/common/src/types/schema.ts [hunk 21]
```diff
@@ -2720,6 +2811,15 @@ export type UpdateStocktakeInput = {
...
 
+export type UpdateStocktakeLineError = {
+  __typename: 'UpdateStocktakeLineError';
+  error: UpdateStocktakeLineErrorInterface;
+};
+
+export type UpdateStocktakeLineErrorInterface = {
+  description: Scalars['String'];
+};
+
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:2822]: no change needed

File: packages/common/src/types/schema.ts [hunk 22]
```diff
@@ -2734,7 +2834,7 @@ export type UpdateStocktakeLineInput = {
   snapshotNumberOfPacks?: InputMaybe<Scalars['Int']>;
 };
 
-export type UpdateStocktakeLineResponse = StocktakeLineNode;
+export type UpdateStocktakeLineResponse = StocktakeLineNode | UpdateStocktakeLineError;
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:2837]: no change needed

File: packages/common/src/types/schema.ts [hunk 23]
```diff
@@ -2765,17 +2865,18 @@ export type UseSuggestedQuantityInput = {
...
 
-export type User = {
-  __typename: 'User';
+export type UserNameDoesNotExist = AuthTokenErrorInterface & {
+  __typename: 'UserNameDoesNotExist';
+  description: Scalars['String'];
+};
+
+export type UserNode = {
+  __typename: 'UserNode';
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:2874]: no change needed

File: packages/common/src/types/schema.ts [hunk 24]
```diff
@@ -2765,17 +2865,18 @@ export type UseSuggestedQuantityInput = {
...
   /** The user's email address */
   email?: Maybe<Scalars['String']>;
   /** Internal user id */
   userId: Scalars['String'];
-};
-
-export type UserNameDoesNotExist = AuthTokenErrorInterface & {
-  __typename: 'UserNameDoesNotExist';
-  description: Scalars['String'];
+  username: Scalars['String'];
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:2879]: 🤔 

Would be cool to have a little popover when you hover a user name which shows details like their email and name etc

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 25]
```diff
@@ -189,16 +189,17 @@ export const useOutboundLines = (
 ): UseQueryResult<OutboundLineFragment[], unknown> => {
   const selectLines = useCallback(
     (invoice: OutboundFragment) => {
+      const forListView = (line: OutboundLineFragment) =>
+        isA.stockOutLine(line) || isA.placeholderLine(line);
       return itemId
         ? invoice.lines.nodes.filter(({ item }) => itemId === item.id)
-        : invoice.lines.nodes.filter(isA.stockOutLine);
+        : invoice.lines.nodes.filter(forListView);
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:196]: Offroad - placeholders being filtered 🙄

**Review by mark-prins — APPROVED** [review]
😁  awesome - thanks! working nicely

File: packages/invoices/src/OutboundShipment/api/hooks.ts (hunk 25 above)
Comment (joshxg) [reply packages/invoices/src/OutboundShipment/api/hooks.ts:196]: It was an odd addition! this is reverts

**Comment by joshxg** [comment]
> the demo server now has a good amount of name_store_joins
> A reasonable number have appeared in my sqlite as well. Sorry.. maybe it's just you?

Yay 😁

**Pull request merged**